IONFileViewerLib/IONFLVWFileDownloader.swift

```diff
@@ -0,0 +1,38 @@
+import Foundation
+
+class IONFLVWFileDownloader {
+    
+    static func downloadFile(url: URL, completion: @escaping (_ success: Bool, _ inner: IONFLVWURLCompletionHandler) -> Void) {
+        let itemUrl = url
+        let documentsDirectoryURL = FileManager.default.temporaryDirectory
+        let fileExtension = url.pathExtension
+        let fileName = url.deletingPathExtension().lastPathComponent
+        let destinationUrl = documentsDirectoryURL
+            .appendingPathComponent(fileName + UUID().uuidString)
+            .appendingPathExtension(fileExtension)
+    
+        let fileResult = IONFLVWFileResult(name: fileName + "." + url.pathExtension, destinationUrl: destinationUrl)
+        
+        if FileManager.default.fileExists(atPath: destinationUrl.path) {
```

Comment:
Could we return early in this `if` statement to avoid the `else` and thus avoid that level of nesting?